### PR TITLE
Update SampleDataUtil and Reminder handling

### PR DESCRIPTION
**Pull request by petermonky**
### This PR:

* updates `SampleDataUtil.java` to better reflect Tinner features
* allows Tinner to save sample data upon start-up
* changes logic to update reminder list upon a call to `StorageManager::saveCompanyList`
  * implementation of a potential `ShowReminderWindowCommand` is rendered trivial
* creates the `DuplicateReminderException`

### Issues that persist are:

* colourcoding of reminders not in place
  * could implement after the application demo

### CODE REVIEW

File: src/main/java/seedu/tinner/model/company/CompanyNameContainsKeywordsPredicate.java [hunk 1]
```diff
@@ -25,16 +24,16 @@ public CompanyNameContainsKeywordsPredicate(List<String> roleNameKeywords, List<
...
-     * Returns true when the {@code Company} contains a {@code Role} witth
+     * Returns true when the {@code Company} contains a {@code Role} with
      * name that matches one of the Strings in {@code roleNameKeywords}
      *
      * @param company {@code Company} to be evaluated
      * @return
      */
     public boolean hasRoleNameKeywords(Company company) {
-        List<Role> roles = company.getRoleManager().getRoleList().getRoles();
+        ReadOnlyRoleList roleList = company.getRoleList();
```
Comment (jinnhl) [inline src/main/java/seedu/tinner/model/company/CompanyNameContainsKeywordsPredicate.java:34]: Good practice of SLAP

File: src/main/java/seedu/tinner/model/reminder/UniqueReminderList.java [hunk 2]
```diff
@@ -91,19 +95,41 @@ public void add(Reminder toAdd) {
...
+     * @param companyList The company from which the reminders are to be produced.
+     */
+    public void setReminders(ObservableList<Company> companyList) {
+        for (Company company : companyList) {
+            ReadOnlyRoleList roleList = company.getRoleList();
+            for (Role role : roleList.getRoleList()) {
+                if (role.getReminderDate().isWithinReminderWindow()) {
+                    Reminder reminder =
+                            new Reminder(company.getName(), role.getName(), role.getStatus(), role.getReminderDate());
+                    add(reminder);
```
Comment (jinnhl) [inline src/main/java/seedu/tinner/model/reminder/UniqueReminderList.java]: Is there any way to get rid of the arrowhead here? Maybe abstracting out the if check on line 124 into another method?

**Review by jinnhl — APPROVED** [review]
Other than the one problem mentioned in the review, implementation wise it looks good, special care was taken for JavaDocs. LGTM 👍

File: src/main/java/seedu/tinner/model/reminder/UniqueReminderList.java (hunk 2 above)
Comment (petermonky) [reply src/main/java/seedu/tinner/model/reminder/UniqueReminderList.java]: Yes that would work, I'll make the changes accordingly. Thanks!

**Pull request merged**